state/endpoint_bindings.go

```diff
@@ -425,26 +435,57 @@ func (b *Bindings) validateForCharm(charmMeta *charm.Meta) error {
 	return nil
 }
 
+// DefaultEndpointBindingSpace returns the current space ID to be used for
+// the default endpoint binding.
+func (st *State) DefaultEndpointBindingSpace() (string, error) {
+	model, err := st.Model()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	cfg, err := model.Config()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	defaultBinding := network.AlphaSpaceId
+
+	space, err := st.SpaceByName(cfg.DefaultSpace())
+	if err != nil && !errors.IsNotFound(err) {
+		return "", errors.Trace(err)
+	}
+	if err == nil {
+		defaultBinding = space.Id()
+	}
+
+	return defaultBinding, nil
+}
+
 // DefaultEndpointBindingsForCharm populates a bindings map containing each
 // endpoint of the given charm metadata (relation name or extra-binding name)
 // bound to an empty space.
-func DefaultEndpointBindingsForCharm(charmMeta *charm.Meta) map[string]string {
+func DefaultEndpointBindingsForCharm(st EndpointBinding, charmMeta *charm.Meta) (map[string]string, error) {
+	defaultBinding, err := st.DefaultEndpointBindingSpace()
```

Comment:
can we rename `defaultBinding` to something like `appDefaultSpaceID`? I think we should be very explicit about what these functions return so as not to confuse IDs with names.

Reply:
done